### ISSUE TITLE
EmbeddedMediaListPlayerComponent is not embedding the video output correctly

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#1127]
Check the BasicEmbeddedMediaListPlayerComponent class in vlcj-examples, it is now for some reason not setting the video surface correctly, suggesting a bug in the component itself - not the test class.

**Comment by caprica** [comment]
For a short term fix that would not require a new release of vlcj, a client application can implement the onAfterConstruct() method described above in their own EmbeddedMediaListComponent sub-class.

**Comment by caprica** [comment]
Did you try the fix from my comment above? When you do "new EmbeddedMediaListPlayerComponent()" you can dynamically sub-class it to implement the "onAfterConstruct()".

It would be very useful to know if that fixes your problem.

**Comment by lsiem** [comment]
```
var embeddedMediaListPlayerComponent = new EmbeddedMediaListPlayerComponent() {
            @Override
            protected void onAfterConstruct() {
                this.mediaListPlayer.mediaPlayer().setMediaPlayer(mediaPlayer());
            }
};

```
Like this?

mediaListPlayer has private access in EmbeddedMediaListPlayerComponent class

**Comment by caprica** [comment]
Indeed the field is private, but there's a public accessor for it:
```
        embeddedMediaListPlayerComponent = new EmbeddedMediaListPlayerComponent() {
            @Override
            protected void onAfterConstruct() {
                mediaListPlayer().mediaPlayer().setMediaPlayer(mediaPlayer());
            }
        };
```
The method invocation admittedly looks a bit weird with so many "mediaPlayer" strings in there.

If this works without any follow-on problems, you will no longer need to do that when a maintenance release of vlcj 4.x is made.

**Comment by caprica** [comment]
By the way, I recommend managing your own playlist rather than doing this, but it should work regardless.

The media list stuff is a bit unloved in LibVLC itself.

**Comment by lsiem** [comment]
All right, thanks for the hint.
After I changed the code accordingly, everything works fine!

**Comment by caprica** [comment]
Resolved at d2771b06a41b064704d641ec0f8d671d467b4196.

Will be part of upcoming 4.7.2 release.

**Comment by caprica** [comment]
4.7.2 released.
